### ISSUE TITLE
Support multiple AD affiliation lines in Medline parser

### ISSUE THREAD

**Issue by guerrajorge** [biopython/biopython#1382]
### Setup

I am reporting a problem with Biopython version, Python version, and operating
system as follows:

```python
import sys; print(sys.version)
import platform; print(platform.python_implementation()); print(platform.platform())
import Bio; print(Bio.__version__)
```

3.6.1 |Anaconda custom (64-bit)| (default, May 11 2017, 13:25:24) [MSC v.1900 64 bit (AMD64)]
CPython
Windows-7-6.1.7601-SP1
1.69

### Expected behavior

`fetch_records_handle = Entrez.efetch(db="pubmed", id=id_list, retmode="text", rettype="medline")
fetch_records = Medline.parse(fetch_records_handle)  

for record in fetch_records:
    authors = record.get("FAU")
    affiliation = record.get("AD")`

`print(authors)`
['Fennelly, Colin', 'Amaravadi, Ravi K']
`print(affiliation)`
['Department of Medicine and Abramson Cancer Center, Perelman School of Medicine, University of Pennsylvania, 777 South Tower PCAM, 34th St. and Civic Center Blvd., Philadelphia, PA, 19104, USA.', 
 'Department of Medicine and Abramson Cancer Center, Perelman School of Medicine, University of Pennsylvania, 777 South Tower PCAM, 34th St. and Civic Center Blvd., Philadelphia, PA, 19104, USA. ravi.amaravadi@uphs.upenn.edu.'

### Actual behavior

`print(authors)`
['Fennelly, Colin', 'Amaravadi, Ravi K']
`print(affiliation)`
'Department of Medicine and Abramson Cancer Center, Perelman School of Medicine, University of Pennsylvania, 777 South Tower PCAM, 34th St. and Civic Center Blvd., Philadelphia, PA, 19104, USA. Department of Medicine and Abramson Cancer Center, Perelman School of Medicine, University of Pennsylvania, 777 South Tower PCAM, 34th St. and Civic Center Blvd., Philadelphia, PA, 19104, USA. ravi.amaravadi@uphs.upenn.edu.'
### Steps to reproduce

Is there a way to get a list from the affiliation, if more than one, just like we get a list from the authors?

**Comment by peterjc** [comment]
Can you save the raw MedLine formatted text file and share it - or at least tell us a couple of the pmid IDs with the problem so we can try to reproduce this? Thanks.

**Comment by guerrajorge** [comment]
Hello Peter,

I see the problem with these PMIDs: 28456991, 28456694. 

Thank you for your help!

**Comment by peterjc** [comment]
Thanks, here are the raw records in MedLine format:

```
PMID- 28456991
OWN - NLM
STAT- In-Process
DA  - 20170430
LR  - 20170805
IS  - 1940-6029 (Electronic)
IS  - 1064-3745 (Linking)
VI  - 1594
DP  - 2017
TI  - Lysosomal Biology in Cancer.
PG  - 293-308
LID - 10.1007/978-1-4939-6934-0_19 [doi]
AB  - Cells depend on the lysosome for sequestration and degradation of macromolecules 
      in order to maintain metabolic homeostasis. These membrane-enclosed organelles
      can receive intracellular and extracellular cargo through endocytosis,
      phagocytosis, and autophagy. Lysosomes establish acidic environments to activate 
      enzymes that are able to break down biomolecules engulfed through these various
      pathways. Recent advances in methods to study the lysosome have allowed the
      discovery of extended roles for the lysosome in various diseases, including
      cancer, making it an attractive and targetable node for therapeutic intervention.
      This review focuses on key aspects of lysosomal biology in the context of cancer 
      and how these properties can be exploited for the development of new therapeutic 
      strategies. This will provide a contextual framework for how advances in
      methodology could be applied in future translational research.
FAU - Fennelly, Colin
AU  - Fennelly C
AD  - Department of Medicine and Abramson Cancer Center, Perelman School of Medicine,
      University of Pennsylvania, 777 South Tower PCAM, 34th St. and Civic Center
      Blvd., Philadelphia, PA, 19104, USA.
FAU - Amaravadi, Ravi K
AU  - Amaravadi RK
AD  - Department of Medicine and Abramson Cancer Center, Perelman School of Medicine,
      University of Pennsylvania, 777 South Tower PCAM, 34th St. and Civic Center
      Blvd., Philadelphia, PA, 19104, USA. ravi.amaravadi@uphs.upenn.edu.
LA  - eng
GR  - P01 CA114046/CA/NCI NIH HHS/United States
GR  - P30 CA016520/CA/NCI NIH HHS/United States
GR  - R01 CA169134/CA/NCI NIH HHS/United States
PT  - Journal Article
PL  - United States
TA  - Methods Mol Biol
JT  - Methods in molecular biology (Clifton, N.J.)
JID - 9214969
PMC - PMC5542621
MID - NIHMS883816
OTO - NOTNLM
OT  - Autophagy
OT  - Cancer
OT  - Drug therapy
OT  - Lysosome
EDAT- 2017/05/01 06:00
MHDA- 2017/05/01 06:00
CRDT- 2017/05/01 06:00
AID - 10.1007/978-1-4939-6934-0_19 [doi]
PST - ppublish
SO  - Methods Mol Biol. 2017;1594:293-308. doi: 10.1007/978-1-4939-6934-0_19.

PMID- 28456694
OWN - NLM
STAT- Publisher
DA  - 20170430
LR  - 20170708
IS  - 1934-1563 (Electronic)
IS  - 1934-1482 (Linking)
DP  - 2017 Apr 26
TI  - Predictive Indices for Functional Improvement and Deterioration,
      Institutionalization, and Death Among Elderly Medicare Beneficiaries.
LID - S1934-1482(16)31258-8 [pii]
LID - 10.1016/j.pmrj.2017.04.005 [doi]
AB  - BACKGROUND: Prediction models can help clinicians provide the best and most
      appropriate care to their patients and can help policy makers design services for
      groups at highest risk for poor outcomes. OBJECTIVE: To develop prediction models
      identifying both risk factors and protective factors for functional
      deterioration, institutionalization, and death. DESIGN: Cohort study using data
      from the Medicare Current Beneficiary Survey (MCBS) SETTING: Community survey.
      PARTICIPANTS: This study included 21,264 Medicare beneficiaries 65 years of age
      and older who participated in the MCBS from the 2001-2008 entry panels and were
      followed up for 2 years. METHODS: The index was derived in 60% and validated in
      the remaining 40%. beta Coefficients from a multinomial logistic regression model
      were used to derive points, which were added together to create scores associated
      with the outcome. MAIN OUTCOME MEASURE: The outcome was activity of daily living 
      (ADL) stage transitions over 2 years following entry into the MCBS. Beneficiaries
      were categorized into 1 of 4 outcome categories: stable or improved function,
      functional deterioration, institutionalization, or death. RESULTS: Our model
      identified 16 factors for functional deterioration (age, gender, education,
      living arrangement, dual eligibility, proxy use, Alzheimer disease/dementia,
      angina pectoris/coronary heart disease, diabetes, emphysema/asthma/chronic
      obstructive pulmonary disease, mental/psychiatric disorder, Parkinson disease,
      stroke/brain hemorrhage, hearing impairment, vision impairment, and baseline ADL 
      stage) after backward selection (P < .05). Compared to stable or improved
      function, the risk of functional deterioration ranged from </=1 to >/=6, </=4 to 
      >/=22 for the risk of institutionalization, and </=3 to >/=16 for the risk of
      death. CONCLUSION: Predictive indices, or point and scoring systems used to
      predict outcomes, can identify elderly Medicare beneficiaries at risk for
      functional deterioration, institutionalization, and death and can aid policy
      makers, clinicians, and family members in improving care for older adults and
      supporting successful aging in the community. LEVEL OF EVIDENCE: I.
CI  - Copyright (c) 2017 American Academy of Physical Medicine and Rehabilitation.
      Published by Elsevier Inc. All rights reserved.
FAU - Kurichi, Jibby E
AU  - Kurichi JE
AD  - Center for Clinical Epidemiology and Biostatistics, University of Pennsylvania,
      Perelman School of Medicine, 423 Guardian Dr, 907 Blockley Hall, Philadelphia, PA
      19104-6021. Electronic address: jkurichi@mail.med.upenn.edu.
FAU - Kwong, Pui L
AU  - Kwong PL
AD  - Center for Clinical Epidemiology and Biostatistics, Perelman School of Medicine, 
      University of Pennsylvania, Philadelphia, PA.
FAU - Xie, Dawei
AU  - Xie D
AD  - Center for Clinical Epidemiology and Biostatistics, Perelman School of Medicine, 
      University of Pennsylvania, Philadelphia, PA.
FAU - Bogner, Hillary R
AU  - Bogner HR
AD  - Center for Clinical Epidemiology and Biostatistics, Perelman School of Medicine, 
      University of Pennsylvania, Philadelphia, PA; Department of Family Medicine and
      Community Health, Perelman School of Medicine, University of Pennsylvania,
      Philadelphia, PA.
LA  - eng
GR  - R01 MH082799/MH/NIMH NIH HHS/United States
PT  - Journal Article
DEP - 20170426
PL  - United States
TA  - PM R
JT  - PM & R : the journal of injury, function, and rehabilitation
JID - 101491319
EDAT- 2017/05/01 06:00
MHDA- 2017/05/01 06:00
CRDT- 2017/05/01 06:00
PHST- 2016/12/09 [received]
PHST- 2017/04/06 [accepted]
AID - S1934-1482(16)31258-8 [pii]
AID - 10.1016/j.pmrj.2017.04.005 [doi]
PST - aheadofprint
SO  - PM R. 2017 Apr 26. pii: S1934-1482(16)31258-8. doi: 10.1016/j.pmrj.2017.04.005.
```

Generated with:

```
$ python3
fPython 3.6.0 (v3.6.0:41df79263a11, Dec 22 2016, 17:23:13) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from Bio import Entrez
>>> Entrez.email = "p.j.a.cock@googlemail.com"
>>> with open("medline.txt", "w") as out_handle:
...     out_handle.write(Entrez.efetch(db="pubmed", id=["28456991", "28456694"], retmode="text", rettype="medline").read())
... 
6685
```

I can then reproduce the issue with the saved medline records,

```
>>> from Bio import Medline
>>> with open("medline.txt") as handle:
...     for record in Medline.parse(handle):
...        print(record.get("FAU"))
...        print(record.get("AD"))
... 
['Fennelly, Colin', 'Amaravadi, Ravi K']
Department of Medicine and Abramson Cancer Center, Perelman School of Medicine, University of Pennsylvania, 777 South Tower PCAM, 34th St. and Civic Center Blvd., Philadelphia, PA, 19104, USA. Department of Medicine and Abramson Cancer Center, Perelman School of Medicine, University of Pennsylvania, 777 South Tower PCAM, 34th St. and Civic Center Blvd., Philadelphia, PA, 19104, USA. ravi.amaravadi@uphs.upenn.edu.
['Kurichi, Jibby E', 'Kwong, Pui L', 'Xie, Dawei', 'Bogner, Hillary R']
Center for Clinical Epidemiology and Biostatistics, University of Pennsylvania, Perelman School of Medicine, 423 Guardian Dr, 907 Blockley Hall, Philadelphia, PA 19104-6021. Electronic address: jkurichi@mail.med.upenn.edu. Center for Clinical Epidemiology and Biostatistics, Perelman School of Medicine, University of Pennsylvania, Philadelphia, PA. Center for Clinical Epidemiology and Biostatistics, Perelman School of Medicine, University of Pennsylvania, Philadelphia, PA. Center for Clinical Epidemiology and Biostatistics, Perelman School of Medicine, University of Pennsylvania, Philadelphia, PA; Department of Family Medicine and Community Health, Perelman School of Medicine, University of Pennsylvania, Philadelphia, PA.
```

Referring to https://www.nlm.nih.gov/bsd/mms/medlineelements.html#ad it seems multiple ``AD`` lines was added in 2014, prior to that only the first author details were provided. Our current test cases only have one ``AD`` line per record, and I presume the parser still expects this.

**Comment by peterjc** [comment]
@mdehoon could you look at this please? I'm not sure what the best solution would be - perhaps returning a list of strings, although that would break existing scripts?

**Comment by mdehoon** [comment]
Given that Medline records can now contain multiple AD lines, I think that the parser should now return a list of strings. Looking at the parser code, I think that treating AD the same way as MH should solve the problem.
@guerrajorge  Would you be able to create a pull request to fix this bug?

**Comment by ghsamuel** [comment]
Hi! When parsing pmids with multiple ADs for a single FAU, is there a way to denote or distinguish which FAU the AD belongs to? For example after parsing I get 6 authors and 9 AD because some authors have nultiple AD. I am trying to make a file with FAU and the right AD. Any help will be greatly appreciated! Thanks

**Comment by peterjc** [comment]
The early examples had one affiliation per author, and if that was always true we'd get two lists with the same number of entries. But that assumption does not hold (anymore?).

New issue #4303 is essentially the same point raised by @ghsamuel, with a specific example https://pubmed.ncbi.nlm.nih.gov/37195739/?format=pubmed

```
...
FAU - Oh, Mijung
AU  - Oh M
AD  - Department of Pathology, School of Medicine.
FAU - Batty, Skylar
AU  - Batty S
AD  - Undergraduate Pipeline Network Summer Research Program, University of New Mexico 
      Health Sciences Center.
AD  - Department of Molecular and Cellular Biology, University of Arizona, Tucson, AZ 
      85721, USA.
FAU - Banerjee, Nayan
AU  - Banerjee N
AD  - School of Chemical Sciences, Indian Association for the Cultivation of Science, 
      2A & 2B Raja S. C. Mullick Road, Jadavpur, Kolkata 700032, West Bengal, India.
FAU - Kim, Tae-Hyung
AU  - Kim TH
AD  - Department of Pathology, School of Medicine.
AD  - University of New Mexico Comprehensive Cancer Center, Albuquerque, NM 87131, USA.
...
```

Four authors (four FAU lines), with 1, 2, 1 and 2 affiliations respectively (six AD lines).

Because we currently parse the information into two separate lists with no mapping between them, we don't track which of the 6 affiliation(s) belong to which of the 4 authors.

I suggest we leave this issue closed, and continue with new issue #4303.
